### PR TITLE
Cleanup docstrings related to interactive mode

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -1137,7 +1137,14 @@ def interactive(b):
 
 
 def is_interactive():
-    """Return whether to redraw after every plotting command."""
+    """
+    Return whether to redraw after every plotting command.
+
+    .. note::
+
+        This function is only intended for use in backends. End users should
```
Comment (timhoffm) [inline lib/matplotlib/__init__.py:1145]: As far as I understand this, interactive mode does not make sense without `pyplot` (without a figure manager no updates are done anyway).

File: lib/matplotlib/__init__.py [hunk 2]
```diff
@@ -1137,7 +1137,14 @@ def interactive(b):
...
+        use `.pyplot.isinteractive` instead.
```
Comment (timhoffm) [inline lib/matplotlib/__init__.py:1146]: As far as I understand this, interactive mode does not make sense without pyplot (without a figure manager no updates are done anyway).

File: doc/devel/contributing.rst [hunk 3]
```diff
@@ -354,9 +354,7 @@ The definition of the pylab text function is a simple pass-through to
 
   # in pylab.py
   def text(*args, **kwargs):
-      ret = gca().text(*args, **kwargs)
-      draw_if_interactive()
-      return ret
+      return gca().text(*args, **kwargs)
```
Comment (timhoffm) [inline doc/devel/contributing.rst:357]: This was outdated. There is no explicit `draw_if_interactive()` involved anymore.

File: lib/matplotlib/__init__.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/__init__.py:1146]: I suppose this should indicate why, and the one in `pyplot` should also link back here for backends.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**